New website: review from [2/3] to [3/3]: New Pipelines

ThomasDelteil opened the pull request:
Review the new pipeline in isolation here

ThomasDelteil committed:
Updating the CI for building the new website

Co-authored-by: Aaron Markham <markhama@amazon.com>
Co-authored-by: Thomas Delteil <thomas.delteil1@gmail.com>

ThomasDelteil committed:
final tweaks

Co-authored-by: Thomas Delteil <thomas.delteil1@gmail.com>
Co-authored-by: Aaron Markham <markhama@amazon.com>

aaronmarkham committed:
pr validation functions for website

aaronmarkham committed:
fix s3 sync folder

aaronmarkham committed:
adjust folders for s3 publish

aaronmarkham committed:
reactivate doc sets

aaronmarkham committed:
Merge pull request #7 from ThomasDelteil/new_website_pipeline_2_aaron

pr validation functions for website

aaronmarkham committed:
remove dupe file

aaronmarkham committed:
R pdf docs generation

aaronmarkham committed:
Merge pull request #8 from aaronmarkham/new_website_pipeline_2_aaron_rdocs

R pdf creation

aaronmarkham committed:
activate r publishing; switch to archive for final site tar

aaronmarkham committed:
Merge pull request #9 from aaronmarkham/new_website_pipeline_2_aaron_rdocs

activate r publishing; switch to archive for final site tar

ThomasDelteil committed:
fixing links and url + css

ThomasDelteil committed:
fixing tutorials

aaronmarkham committed:
update readmes for doc; remove unused files

ThomasDelteil committed:
fixing some links

ThomasDelteil committed:
fixing more links

ThomasDelteil committed:
fixing jekyll docker image

aaronmarkham committed:
beta deployment script for CI publish job

aaronmarkham committed:
Merge pull request #10 from ThomasDelteil/new_website_pipeline_2_updates

update readmes for doc; remove unused files

ThomasDelteil committed:
Fixing css and js

ThomasDelteil committed:
Merge commit '50ffbd00e144b6a1b75f5d6469debae7753d8c03' into new_website_pipeline_2

vishaalkapoor commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+
+
+// This is for the website and the Python API only
+def docs_prepare_python_only() {
```
Thanks!

It'd be ideal to have some more description in the comments on these methods as it's unclear out of context how things would work.    Also the distinction b/w jekyll and non-jekyll will be lost over time, so a comment will help.

aaronmarkham replied:
> Thanks!
> 
> It'd be ideal to have some more description in the comments on these methods as it's unclear out of context how things would work. Also the distinction b/w jekyll and non-jekyll will be lost over time, so a comment will help.

I'll add a bunch of comments as part of the review.

vishaalkapoor commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+              build 'test-website-publish-master'
+            }
+            catch (Exception e) {
+               println(e.getMessage())
```
println("Building docs failed", e.getMessage()) or something to that effect (syntax there is incorrect).

aaronmarkham replied:
> println("Building docs failed", e.getMessage()) or something to that effect (syntax there is incorrect).

Did you run it to find an error? Or are you a walking groovy interpreter? :)
It's not needed, so the try... catch can be removed. It just made testing a lot easier. I think it was working just fine... but I could be wrong...

vishaalkapoor commented on ci/jenkins/Jenkinsfile_website_c_docs:
```diff
@@ -16,39 +16,37 @@
 
 // timeout in minutes
-max_time = 120
+max_time = 20
```
Is this tested on a similar or slower instance type as in the prod pipeline

aaronmarkham replied:
> Is this tested on a similar or slower instance type as in the prod pipeline

It should run quickly. There's no reason for it to take 120 minutes. But if we find CI is bogged down, I suppose it could timeout in that situation. I say leave until we hit the limit... that way we know something is causing the system to slow.

vishaalkapoor commented:
I'm familiar with the md changes and have reviewed the Jenkins pipeline code. LGTM.

reminisce committed:
Add __array_function__

Add test

Fix

Fix sanity

Fix build failure

Use array_function protocol only when np.version >= 1.17

Add unit tests for array ufunc protocol

Fix pylint

Reformat

Fix build

Fix

Refactor test suite for numpy interoperability

anirudh2290 commented on ci/docker/runtime_functions.sh:
```diff
@@ -1437,9 +1408,404 @@ deploy_docs() {
+        DEV=1                         \
+        USE_CPP_PACKAGE=1             \
+        USE_BLAS=openblas             \
+        USE_MKLDNN=0                  \
```
why USE_MKLDNN=0 ?

ThomasDelteil replied:
faster build

anirudh2290 commented on ci/docker/runtime_functions.sh:
```diff
@@ -1437,9 +1408,404 @@ deploy_docs() {
+
+
+build_docs_setup() {
+    build_folder="docs/_build"
```
no set -ex for this step ?

ThomasDelteil replied:
good point

anirudh2290 commented on ci/docker/runtime_functions.sh:
```diff
@@ -1437,9 +1408,404 @@ deploy_docs() {
+
+    popd
+
+    GZIP=-9 tar zcvf $artifacts_path $docs_build_path
```
what does this line do ?

ThomasDelteil replied:
this zips up with high compression (GZIP=-9) the generated artifact in a specific path so that it can then be picked up by CI for building the entire website as the sum of its components

anirudh2290 commented on ci/docker/runtime_functions.sh:
```diff
@@ -1437,9 +1408,404 @@ deploy_docs() {
+
+build_version_doc() {
+
+   # $1 is the list of branch or tag to build
```
no set -ex here ?

ThomasDelteil replied:
good point

aaronmarkham replied:
This function can be deleted. It's only used for the current website.

anirudh2290 commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
 }
 
+
+def compile_unix_lite() {
```
How much time are these sstages going to add to the CI. Will they be run as part of every PR or nightly ?

aaronmarkham replied:
> How much time are these sstages going to add to the CI. Will they be run as part of every PR or nightly ?

They run in parallel, so they're actually shorter than what we have now.
It is setup to run with every PR.

anirudh2290 commented on docs/python_docs/python/tutorials/extend/customop.md:
```diff
@@ -0,0 +1,218 @@
+# Creating custom operators with numpy
```
@ptrendx pointed out that AMP docs are missing

ThomasDelteil replied:
that's correct, some tutorials are missing and will be part of a post-launch re-alignment

aaronmarkham commented on Makefile:
```diff
@@ -718,6 +704,10 @@ rpkgtest:
 	Rscript -e 'res<-covr:::package_coverage("R-package");fileConn<-file(paste("r-package_coverage_",toString(runif(1)),".json"));writeLines(covr:::to_codecov(res), fileConn);close(fileConn)'
 
+
+sample_lib:
```
What's this for?

ThomasDelteil replied:
was removed by mistake, is required for build

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+      node(NODE_LINUX_CPU) {
+        ws('workspace/docs') {
+          timeout(time: max_time, unit: 'MINUTES') {
+            archiveArtifacts 'docs/_build/full_website.tgz'
```
This should be replaced with upload to S3 once we have a working CI-dev env for it.

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+            //sh 'tar -xzf docs/_build/full_website.tgz --directory .'
+            // TODO: Make sure this 'test-website-publish' understand the new structure
+            try {
+              build 'test-website-publish-master'
```
Needs to be updated to production.

```suggestion
              build 'website-publish-master'
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -112,7 +112,7 @@ def compile_unix_cpu_openblas() {
             utils.init_git()
             utils.docker_run('ubuntu_cpu', 'build_ubuntu_cpu_openblas', false)
-            utils.pack_lib('cpu', mx_lib_cython, true)
+            utils.pack_lib('cpu', mx_lib, true)
```
I'm not so sure about this update. Discuss?

ThomasDelteil replied:
that's what is used for the unpacking

aaronmarkham replied:
But why mx_lib instead of mx_lib_cython? I had made some of these updates to improve the speed of the runs, but this function is used by other parts of CI and since we've created our own slimmed down binary generation steps.

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1369,16 +1369,20 @@ def test_qemu_armv7_cpu() {
 }
 
+// This is for running on PRs
 def docs_website() {
```
I think we can delete this function if we're not supporting the current website publish flow. In order for it to be useful, it would need not only S3 upload, but also some kind of web serving. As it is, this will just upload a site that S3 can't server properly.

ThomasDelteil replied:
ok

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
     }]
 }
 
+
```
```suggestion

// This creates the MXNet binary needed for generating different docs sets
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_python() {
```
```suggestion
// Call this function from Jenkins to generate just the Python API microsite artifacts.
def docs_python() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_c() {
```
```suggestion
// Call this function from Jenkins to generate just the C and C++ API microsite artifacts.
def docs_c() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_julia() {
```
```suggestion
// Call this function from Jenkins to generate just the Julia API microsite artifacts.
def docs_julia() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_r() {
```
```suggestion
// Call this function from Jenkins to generate just the R API PDF artifact.
def docs_r() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_scala() {
```
```suggestion
// Call this function from Jenkins to generate just the Scala API microsite artifacts.
// It will also generate the Scala package.
def docs_scala() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_java() {
```
```suggestion
// Call this function from Jenkins to generate just the Java API microsite artifacts.
// It will also generate the Scala package.
def docs_java() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_clojure() {
```
```suggestion
// Call this function from Jenkins to generate just the Clojure API microsite artifacts.
// It will also generate the Scala package.
def docs_clojure() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+def docs_jekyll() {
```
```suggestion
// Call this function from Jenkins to generate just the main website artifacts.
def docs_jekyll() {
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+}
+
+
+// This is for publishing the full website
```
```suggestion
// This is for publishing the full website
// Assumes you have run all of the docs generation functions
// Called from Jenkins_website_full and Jenkins_website_full_pr
```

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+            // utils.pack_lib('full_website', 'docs/_build/full_website.tgz', false)
+
+            // archive so the publish pipeline can access the artifact
+            archiveArtifacts 'docs/_build/full_website.tgz'
```
Should be switched to S3 when a test environment is made available on CI-dev.

ThomasDelteil replied:
sure

aaronmarkham commented on ci/jenkins/Jenkins_steps.groovy:
```diff
@@ -1389,6 +1393,238 @@ def docs_website() {
+            utils.docker_run('ubuntu_cpu_jekyll', 'build_docs_small', false)
+
+            // Publish preview to S3
+            sh "cd docs/_build/html && ci/other/ci_deploy_doc.sh ${env.BRANCH_NAME} ${env.BUILD_NUMBER}"
```
Not useful until there's a website serving process attached to S3.

ThomasDelteil replied:
will comment out until you / someone setups an alternative setup

aaronmarkham reviewed (commented):
Some suggested changes and comments.
I'm ignoring the /docs/python_docs and other folders since I don't think they're intended to be in this PR.

ThomasDelteil committed:
Update ci/jenkins/Jenkins_steps.groovy

Co-Authored-By: Aaron Markham <markhama@amazon.com>

ThomasDelteil committed:
Update ci/jenkins/Jenkins_steps.groovy

Co-Authored-By: Aaron Markham <markhama@amazon.com>

ThomasDelteil committed:
Update ci/jenkins/Jenkins_steps.groovy

Co-Authored-By: Aaron Markham <markhama@amazon.com>

ThomasDelteil committed:
Update ci/jenkins/Jenkins_steps.groovy

Co-Authored-By: Aaron Markham <markhama@amazon.com>

ThomasDelteil committed:
Apply suggestions from code review

Co-Authored-By: Aaron Markham <markhama@amazon.com>

ThomasDelteil committed:
merging changes to license headers into last PR

aaronmarkham committed:
update to reserved instances to publish job

aaronmarkham committed:
fix conditions for PR tests

ThomasDelteil committed:
fix python bug

ThomasDelteil committed:
Merge pull request #14 from ThomasDelteil/new_website_pipeline_2_test_pr

fix conditions for PR tests

ElaineBao committed:
avoid test relu at the origin due to discontinuous gradient (#16133)

* avoid test relu at the origin due to discontinuous gradient

* retrigger CI

aaronmarkham committed:
Merge branch 'new_website_pipeline_2' of https://github.com/ThomasDelteil/incubator-mxnet into new_website_pipeline_2_restricted

ThomasDelteil committed:
merging branch #2 in #3

ThomasDelteil committed:
Merge branch 'new_website_delete_content' into new_website_pipeline_2

ThomasDelteil committed:
update submodule to crrect value

ThomasDelteil committed:
reverse wrong update

ThomasDelteil committed:
remove wrong copy statement

ThomasDelteil committed:
Merge pull request #15 from ThomasDelteil/new_website_pipeline_2_restricted

use restricted nodes for publish job

aaronmarkham committed:
add license header

ThomasDelteil committed:
linting of cpp in R

aaronmarkham committed:
add placeholder for website ci check

aaronmarkham committed:
fix conflict

reminisce committed:
Fix remaining errors reported by D2L (#16157)

* Fix

* Fix ssd

* Add test for np.empty

* Add np.linalg.norm

* Fix indexing bug

* Improve doc

hzfan committed:
[Numpy] Numpy copysign (#15851)

* add numpy compatible copysign

* fix scalar op registration error

* add test

ZhennanQin committed:
Update dmlc-core (#16149)

Change-Id: I472fb7bbffc16ed8c36494ab49838b08c59b2f12

sxjscience committed:
use 1E-4 in groupnorm test(#16169)

tingying2020 committed:
numpy operator ravel, derive from reshape (#16016)

* it is the same as reshape(x, -1)

* register reshape with prefix _npi_

* fix format error

* edit examples in doc

* fix error in review

* add out in wrapper

* remove out

* test data type and add order

* change order check

* remove redundant test and add docstring about order

perdasilva committed:
Adds dynamic libmxnet to CD pipeline (#16163)

* Adds dynamic libmxnet release pipeline

* Updates default variants on CD pipeline

ChaiBapchya committed:
Sequence last fix (#16156)

* seq last fix

* index tensor to have int64

* fix dtypes

* revert unnecessary changes

* if seq len not passed, pass int64 dtype

* dtype comment

* use int32 or int64 as index dtype based on build flag

* Trigger notification

* Trigger notification

* lint fix

Vikas-kum committed:
fixing test for model compatibility checker (#16159)

* fixing test for model compatibility checker

* fake change

aaronmarkham committed:
add mock

Vikas-kum committed:
Tutorials nighly fix (#16179)

* Fixing tutorials test failures

* profier set state

* modifying Jenkins for fast run

* chanign comments

* more test comment for faster run

* changing comments from

* more comments

* profiler tutorial test fixes

* uncommenting all tutorial tests

* enabling all tests for nightly

* removing parallel keyword from tutorials test

* updating comments

* Added github issue link in the comment

aaronmarkham committed:
restore python deps

aaronmarkham committed:
consolidate python deps

aaronmarkham committed:
copy requirements

aaronmarkham committed:
add requirements file

aaronmarkham committed:
add python back

aaronmarkham committed:
revert julia dockerfile

aaronmarkham committed:
add requirements file

aaronmarkham committed:
rename requirements file

aaronmarkham committed:
update doc on requirements file

aaronmarkham committed:
Merge pull request #16 from ThomasDelteil/new_website_pipeline_2_mock

consolidate python deps

aaronmarkham committed:
Merge branch 'new_website_pipeline_2' of https://github.com/ThomasDelteil/incubator-mxnet into new_website_pipeline_2_syntax_error

aaronmarkham committed:
fix syntax error... hopefully

ThomasDelteil committed:
Merging conflicts, removing nightly tutorial tests

ThomasDelteil committed:
Update Jenkins_steps.groovy

aaronmarkham committed:
add missing req file

ThomasDelteil committed:
Update Jenkins_steps.groovy

aaronmarkham committed:
Merge pull request #18 from ThomasDelteil/new_website_pipeline_2_mock

add missing req file

ThomasDelteil committed:
Update Jenkins_steps.groovy

reminisce committed:
Improved error mesages

ThomasDelteil committed:
Update Jenkins_steps.groovy

ThomasDelteil committed:
Merge pull request #17 from ThomasDelteil/new_website_pipeline_2_syntax_error

fix syntax error

ThomasDelteil committed:
update dmlc core

ThomasDelteil committed:
Merge branch 'new_website_pipeline_2' of https://github.com/ThomasDelteil/incubator-mxnet into new_website_pipeline_2

reminisce committed:
Fix np.choice

anirudh2290 committed:
Add register_op_hook for gluon (#15839)

* Set monitor callback basic support

* Trigger CI

* Add base.pyi and ndarray.pyx

* Change not supported to experimental and check for both static_shape and static_alloc

aaronmarkham committed:
adding codeowners (#16165)

* add aaron to codeowners

* retrigger CI

* ci

kshitij12345 committed:
assert_allclose -> rtol=1e-10 (#16198)

eric-haibin-lin committed:
[Dataset] add shard API (#16175)

* add shard API

* more balanced

* fix a bug

* Update test_gluon_data.py

ThomasDelteil committed:
removing the utility nodes

ThomasDelteil committed:
use restricted nodes for everything

szha committed:
[MEMORY] retry GPU memory allocation if fragmented (#16194)

zachgk committed:
Fix README Build Status (#16183)

* Fix README Build Status

* Add labels to Jenkins Badges

* Add more line breaks

ThomasDelteil committed:
empty commit

ThomasDelteil committed:
empty commit

sxjscience committed:
add exception check for numpy reshape (#16180)

ThomasDelteil committed:
fix typo

ThomasDelteil committed:
fix the restricted-utility bug

ThomasDelteil committed:
deleting unused function

zhreshold committed:
improve dataloader signals and messages (#16114)

* improve dataloader signals and messages

* address comments

* fix spawn tests on windows

xidulu committed:
[Numpy] Numpy behavior normal distribution (#16109)

* normal implemented

* numpy behavior normal imlemented

* retrigger CI

* retrigger CI

* regrigger ci

* add normal parameter check

* add raise for normal

* remove dead code

stu1130 committed:
fix multinomial bug on gpu (#16204)

eric-haibin-lin committed:
Update ndarray.py (#16205)

szha committed:
subscribe to build and CD changes (#16192)

ZhennanQin committed:
Float64 fallback for mkldnn subgraph and rnn op (#15853)

* Fix mkldnn subgraph with float64

* Fix ci

Change-Id: I0bb4e8d7a0a534aa661601887cc633cb9c4fcadf

* Fix test

Change-Id: I96c529abe7adb6def90a22f03b3432263ef12fda

* Update dmlc-core

Change-Id: I472fb7bbffc16ed8c36494ab49838b08c59b2f12

* pin to offical dmlc

Change-Id: I5a27dc83b892bf8fcb34bb089449d1d3b6e9beed

* Fix GPU CI

Change-Id: I285947e01bdb0651c2c7830ed4eb76931a09b754

* Fix GPU CI

Change-Id: I6f23b51d6bda44f6ae18766ebe390118740bb9c7

ThomasDelteil committed:
New Website: Remove Old Content [2/3] (#15885)

* Adding new website files

Co-authored-by: Thomas Delteil <thomas.delteil1@gmail.com>
Co-authored-by: Aaron Markham <markhama@amazon.com>

* removing old content

Co-authored-by: Thomas Delteil <thomas.delteil1@gmail.com>
Co-authored-by: Aaron Markham <markhama@amazon.com>

* add license header

* fixing links and url + css

* fixing tutorials

* exclude python_docs folder when building prod site

This allows things to coexist until a following PR deletes this file.

* license header fixes

* exclude static site from license header check

* exclude static site and sphinx theme

* exclude static_site and cpp_docs from sphinx

* add licenses and excludes

* fix headers

* restore file for legacy builds

* empty commit to trigger CI

* update submodule to correct one

* delete website check

* add placeholder website check

* add placeholder website check, for reals this time

* trigger CI

* trigger PR

ThomasDelteil committed:
merging master
